### PR TITLE
fix(ActionList): last section divider

### LINKED ISSUE #1610: ActionListSection has unwanted divider at the end
ActionListSection has an unwanted divider at the end.
Screenshot 2023-09-07 at 5 10 46 PM

### PR DESCRIPTION
**Pull request by saurabhdaware**
Resolves #1610 

Adding role separator wouldn't work on Divider since divider component doesn't accept the role property. This made the css selector for last actionlist section divider invalid. There was also additional div around divider.

https://github.com/razorpay/blade/blob/2fd09500feefd32a6237be48e88c3874f4f588c6/packages/blade/src/components/ActionList/styles/StyledListBoxWrapper.web.tsx#L12

```js
<Divider
  {...makeAccessible({ role: 'separator' })
/>
```

I have added `role="separator"` to Divider component by default since that is closer to the `<hr />` behaviour https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role

Also removed ActionListSectionDivider component as we don't need that anymore. I'll maybe add some snapshot tests later to test out this scenario. Created issue for that - #1611 (just finishing up AutoComplete right now so dont want to spend too much time on this task)

You can try the change on - InternalSectionListPerformance story

### CODE REVIEW

File: packages/blade/src/components/ActionList/ActionListItem.tsx [hunk 1]
```diff
@@ -139,11 +125,7 @@ const _ActionListSection = ({
       >
         {children}
       </BaseBox>
-      {_hideDivider && isReactNative() ? null : (
-        <Box marginX="spacing.3" marginY="spacing.1">
-          <ActionListSectionDivider />
-        </Box>
-      )}
+      {_hideDivider && isReactNative() ? null : <Divider marginX="spacing.3" marginY="spacing.1" />}
```
Comment (anuraghazra) [inline packages/blade/src/components/ActionList/ActionListItem.tsx:128]: 🤔 how did this fix the bug? I can see no logic changes.

File: packages/blade/src/components/Card/__tests__/__snapshots__/Card.native.test.tsx.snap [hunk 2]
```diff
@@ -1092,6 +1093,7 @@ exports[`<Card /> should render a Card with Header 1`] = `
           </View>
         </View>
         <View
+          accessible={true}
```
Comment (anuraghazra) [inline packages/blade/src/components/Card/__tests__/__snapshots__/Card.native.test.tsx.snap:1096]: How is Card's snapshot updating 🤔

Comment (anuraghazra) [reply packages/blade/src/components/Card/__tests__/__snapshots__/Card.native.test.tsx.snap:1096]: Also this should not be accessible={true}

File: packages/blade/src/components/Divider/Divider.tsx [hunk 3]
```diff
@@ -94,6 +95,9 @@ const Divider = ({
       {...borderColor}
       {...metaAttribute({ name: MetaConstants.Divider, testID })}
       {...getStyledProps(styledProps)}
+      {...makeAccessible({
```
Comment (anuraghazra) [inline packages/blade/src/components/Divider/Divider.tsx:98]: I see now why we are getting accessible={true} 

Let's explicitly override this prop in RN to be false. Otherise dividers will unnecessarily receive focus.

**Review by anuraghazra — CHANGES_REQUESTED** [review]
Mostly looks good, minor questions.

File: packages/blade/src/components/Divider/Divider.tsx (hunk 3 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Divider/Divider.tsx]: ahh got it. I removed the makeAccessible call itself from react native.

File: packages/blade/src/components/ActionList/ActionListItem.tsx (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/ActionList/ActionListItem.tsx:128]: this didn't fix the bug. This is just refactor to remove additional wrapper and nesting of divider. Adding `role="separator"` fixed the bug. More info in PR description

**Review by anuraghazra — APPROVED** [review]
Looks good to me.

**Comment by saurabhdaware** [comment]
I'll merge this on L1 since its a P0. @snitin315 let me know if you have any comments, I'll take them up as followup PR.

**Review by snitin315 — COMMENTED** [review]
LGTM

**Pull request merged**